src/api/dashboard.js

```diff
@@ -65,8 +65,25 @@ Dashboard.prototype = {
   },
 
   onStateChanged: function (callback) {
-    this._dashboard.widgets._widgetsCollection.bind('change', function () {
-      callback(this.getState(), this.getDashboardURL());
+    var firstLoad = true;
+    var widgets = this._dashboard.widgets._widgetsCollection;
+
+    widgets.bind('change', function () {
+      var triggerCallback = true;
+
+      if (firstLoad) {
+        var widgetsNotLoaded = _.filter(widgets.dataviewModel, function (mdl) {
+          return mdl._pending === true;
+        });
+        console.log(arguments[0], widgetsNotLoaded);
+        if (widgetsNotLoaded.length !== 0) {
+          triggerCallback = false;
+        } else {
+          firstLoad = false;
+        }
+      }
+
+      triggerCallback && callback(this.getState(), this.getDashboardURL());
```

Comment:
I think this approach is error-prone, using private variables and leaving this event listener around since it will modify those temporary variables even after the initial events, which could hamper the expected behavior. Let's not go down this path.

I think part of the problem is related to how this _state_ is handled; The initial state should be given as _attributes_, it doesn't really make sense to set "initial state" right after the model was created. I'd remove the setState method and solve the initial-state problem to avoid events that should not event fire initially.